### PR TITLE
rgw: Do not fetch bucket stats by default upon bucket listing

### PR DESCRIPTION
**Pull request by prallabh**
Fixes: http://tracker.ceph.com/issues/20377
Signed-off-by: Pavan Rallabhandi <PRallabhandi@walmartlabs.com>

### CODE REVIEW

**Comment by rzarzynski** [comment]
The second thing: please correct the commit author's name and e-mail. Try:
```
git config user.name "John Smith"
git config user.email "user@host"
```

File: src/rgw/rgw_rest_swift.h [hunk 1]
```diff
@@ -43,7 +43,7 @@ class RGWListBuckets_ObjStore_SWIFT : public RGWListBuckets_ObjStore {
     return 0;
   }
 public:
-  RGWListBuckets_ObjStore_SWIFT() : need_stats(true) {}
+  RGWListBuckets_ObjStore_SWIFT() : need_stats(false) {}
```
Comment (prallabh) [inline src/rgw/rgw_rest_swift.h]: @rzarzynski Thanks for the quick review. I was referring to http://tracker.ceph.com/issues/11285 where this check `need_stats = (s->format != RGW_FORMAT_PLAIN)` was removed. 

With this patch this is what I see for my curl queries, am I missing anything here.

```
curl -i 'http://localhost:8000/swift/v1?stats=true&format=json' -X GET -H "X-Auth-Token: "
HTTP/1.1 200 OK
X-Timestamp: 1498141604.33927
X-Account-Container-Count: 1
X-Account-Object-Count: 1
X-Account-Bytes-Used: 5872
X-Account-Bytes-Used-Actual: 8192
X-Account-Access-Control: {}
X-Trans-Id: tx0000000000000000037c1-00594bd3a4-101b-default
X-Openstack-Request-Id: tx0000000000000000037c1-00594bd3a4-101b-default
Content-Type: application/json; charset=utf-8
Content-Length: 40
Date: Thu, 22 Jun 2017 14:26:44 GMT


curl -i 'http://localhost:8000/swift/v1?stats=true' -X GET -H "X-Auth-Token: "
HTTP/1.1 200 OK
X-Timestamp: 1498141609.86924
X-Account-Container-Count: 1
X-Account-Object-Count: 1
X-Account-Bytes-Used: 5872
X-Account-Bytes-Used-Actual: 8192
X-Account-Access-Control: {}
X-Trans-Id: tx0000000000000000037c2-00594bd3a9-101b-default
X-Openstack-Request-Id: tx0000000000000000037c2-00594bd3a9-101b-default
Content-Type: text/plain; charset=utf-8
Content-Length: 4
Date: Thu, 22 Jun 2017 14:26:49 GMT
```

Comment (rzarzynski) [inline src/rgw/rgw_rest_swift.h]: I'm afraid that the `stats` parameter is our private extension. Tempest doesn't use it but require the stats to be always present:
```
$ sudo ngrep -W byline port 8000 -d lo

...

########
T 127.0.0.1:33146 -> 127.0.0.1:8000 [AP]
GET /v1/KEY_9775d1b07dc245849dfddf3f64857b34/?format=json HTTP/1.1.
Host: 127.0.0.1:8000.
Accept-Encoding: identity.
connection: close.
X-Auth-Token: gAAAAABZS9cb7Oh41tfiRucOVG88cS7rGnfcgmcz9siBlVZPkoKx5k76vcVbMmMcdc_RdRyAyLrwPe7R5BMUhyh0p5v9HbtisHkZ-rkClWqAT0rK35lr-AilpefEU3r916ajnHpjBDSQ9LSqkj7If3cNAB3Hg_RG458Dqh-8iQXwbMFvSnKca7g.
.

##
T 127.0.0.1:8000 -> 127.0.0.1:33146 [AP]
HTTP/1.1 200 OK.
Content-Length: 196.
X-Timestamp: 1498142492.21979.
X-Account-Container-Count: 6.
X-Account-Object-Count: 0.
X-Account-Bytes-Used: 0.
X-Account-Bytes-Used-Actual: 0.
X-Account-Access-Control: {}.
X-Trans-Id: tx00000000000000000002d-00594bd71c-106b-default.
X-Openstack-Request-Id: tx00000000000000000002d-00594bd71c-106b-default.
Accept-Ranges: bytes.
Content-Type: application/json; charset=utf-8.
Date: Thu, 22 Jun 2017 14:41:32 GMT.
Connection: close.
.

##
T 127.0.0.1:8000 -> 127.0.0.1:33146 [AP]
[{"name":"tempest-a--501653990"},{"name":"tempest-b--1185429765"},{"name":"tempest-c--475862726"},{"name":"tempest-d--118749091"},{"name":"tempest-e--1032378481"},{"name":"tempest-f--1618574056"}]
########
```

Comment (prallabh) [inline src/rgw/rgw_rest_swift.h]: FWIW, the stats is added via http://tracker.ceph.com/issues/4759

**Comment by mattbenjamin** [comment]
@rzarzynski can this be implemented as an extension?

**Comment by rzarzynski** [comment]
@mattbenjamin: I think so. We could introduce a configurable for the default value of `need_stats`.

**Comment by prallabh** [comment]
@rzarzynski So would you want a new config to be introduced with the default value for `stats` set to true? Can you please advise further, thanks.

**Comment by rzarzynski** [comment]
@prallabh: yeah, I think this would be the quicker solution for the perfomance-vs-compliance dilemma.

**Comment by prallabh** [comment]
@rzarzynski can you please check now, thanks!

**Comment by rzarzynski** [comment]
jenkins retest this please (unrelated test failures)

**Comment by prallabh** [comment]
@rzarzynski can you please check now, have updated the commit message as well appropriately, thanks.

**Comment by prallabh** [comment]
@rzarzynski can this now be merged?

**Comment by rzarzynski** [comment]
It will be merged after passing a Teuthology run. I will schedule one soon.

**Comment by rzarzynski** [comment]
Waiting for: http://pulpito.ceph.com/rzarzynski-2017-06-29_11:28:03-rgw-wip-rzarzynski-testing---basic-mira/.

**Comment by prallabh** [comment]
@rzarzynski I see some failures in that run, is that holding this to be merged.

**Comment by rzarzynski** [comment]
@prallabh: no worries, it looks they are unrelated. Just in case I'm scheduling a new run.

**Comment by yuriw** [comment]
http://pulpito.ceph.com/yuriw-2017-07-11_19:43:31-rgw-wip-yuri-testing_2017_7_11-distro-basic-smithi/

**Pull request merged**